Sources/Abstract/Wrapper.swift

```diff
@@ -0,0 +1,31 @@
+/*:
+# Wrapper
```

Comment:
This is really interesting. I'm not sure about the name though. Maybe `Wrapper` is fine, but what do you think about `Newtype` (since this is basically Haskell's newtype)